patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -169,8 +169,28 @@ private static void PortFilesFromXtoY(string superSavePath, string source, strin
 				lastLaunchedTml = (string)lastLaunchedTmlObject;
 		}
 		catch (Exception e) {
-			e.HelpLink = "https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#configjson-corrupted";
-			ErrorReporting.FatalExit($"Attempt to Port from \"{oldFolderPath}\" to \"{newFolderPath}\" aborted, the \"{sourceFolderConfig}\" file is corrupted.", e);
+			if (File.GetLastWriteTime(sourceFolderConfig) > migrationDay) {
+				// If the file was edited recently, we assume that an updated tModLoader edited it. This should ignore modifications made long ago by pre-migration logic tModLoader releases.
+				lastLaunchedTml = BuildInfo.tMLVersion.ToString();
+			}
+			else {
+				e.HelpLink = "https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#configjson-corrupted";
+				ErrorReporting.FatalExit($"Attempt to Port from \"{oldFolderPath}\" to \"{newFolderPath}\" aborted, the \"{sourceFolderConfig}\" file is corrupted.", e);
+			}
+		}
+
+		if (string.IsNullOrEmpty(lastLaunchedTml)) {
+			// If the config.json is missing LastLaunchedTModLoaderVersion entry, we can ask the user. (Most likely the user copied Terraria/config.json over)
+			int result = ErrorReporting.ShowMessageBoxWithChoices(
+				title: "Failed to read config.json configuration file",
+				message: "Your config.json file is incomplete.\n\nPlease select one of the following options and the game will resume loading:\n\nWhat is the highest version of tModLoader that you have launched?",
```

Comment:
This should be localizable.

Reply:
All this occurs before localization is loaded - before save paths are finalized.
There was an issue on this last version as well.

Reply:
We can add manual localization if we really need to later, but we should see this issue a lot less for now